tests/rbr/MakeRegionsTest.java

```diff
@@ -0,0 +1,76 @@
+package rbr;
+
+import java.util.*;
+import org.junit.*;
+import util.*;
+
+public class MakeRegionsTest {
+
+    Set<Character> ip = new HashSet<>();
+    Set<Character> op = new HashSet<>();
+    Set<Vertex> dst = new HashSet<>();
+    private RBR rbr;
+
+    @Before
+    public void setUp() {
+        Graph graph = RegularGraphBuilder.generateGraph(4,4);
+        rbr = new RBR(graph);
+        ip.add('I');
+        ip.add('N');
+        ip.add('E');
+        op.add('N');
+        op.add('S');
+        op.add('W');
+        dst.add(new Vertex("0.1"));
+        dst.add(new Vertex("0.2"));
+        dst.add(new Vertex("0.3"));
+        dst.add(new Vertex("1.0"));
+        dst.add(new Vertex("1.2"));
+        dst.add(new Vertex("1.3"));
+        dst.add(new Vertex("2.0"));
+        dst.add(new Vertex("2.3"));
+        dst.add(new Vertex("3.0"));
+        dst.add(new Vertex("3.1"));
+        dst.add(new Vertex("3.2"));
+        dst.add(new Vertex("3.3"));
+    }
+
+    @Test
+    public void emptyDestinationsCreatesNoRegion() throws Exception {
+        List<Region> teste = rbr.makeRegions(dst, ip, op);
+        Set<Vertex> emptyDst = new HashSet<>();
+        List<Region> createdRegionsWithoutDestinations = rbr.makeRegions(emptyDst, ip, op);
+        Assert.assertTrue(createdRegionsWithoutDestinations.isEmpty());
+    }
+
+    @Test
+    public void createdRegionsHaveSameInputPortsOriginals() throws Exception {
+        for(Region adjustedRegion : rbr.makeRegions(dst, ip, op)) {
+            Assert.assertEquals(adjustedRegion.inputPorts(), ip);
+        }
+    }
+
+    @Test
+    public void createdRegionsHaveSameOutputPortsOriginals() throws Exception {
+        for(Region adjustedRegion : rbr.makeRegions(dst, ip, op)) {
+            Assert.assertEquals(adjustedRegion.outputPorts(), op);
+        }
+    }
+
+    @Test
+    public void createdRegionsHaveNoOutsider() throws Exception {
+        for(Region createdRegion : rbr.makeRegions(dst, ip, op)){
+            Assert.assertTrue(createdRegion.outsiders().isEmpty());
+        }
+    }
+
+    @Test
+    public void createdRegionsHaveSameDestinationsOriginals() throws Exception {
+        Set<Vertex> destinationsCreatedRegions = new HashSet<>();
+        Set<Vertex> originalsDst = new HashSet<>(dst);
+        for(Region createdRegion : rbr.makeRegions(originalsDst, ip, op)) {
+            destinationsCreatedRegions.addAll(createdRegion.destinations());
+        }
+        Assert.assertEquals(destinationsCreatedRegions, dst);
+    }
+}
```

Comment:
You can improve this by putting some test to verify if the number of created regions is correct and if the max and min of regions box is also correct.